app/controllers/api/irs_attempts_api_controller.rb

```diff
@@ -81,7 +81,9 @@ def timestamp
       timestamp_param = params.permit(:timestamp)[:timestamp]
       return nil if timestamp_param.nil?
 
-      Time.strptime(timestamp_param, '%Y-%m-%dT%H:%M:%S%z')
+      date_fmt = timestamp_param.match?('\.') ? '%Y-%m-%dT%H:%M:%S.%N%z' : '%Y-%m-%dT%H:%M:%S%z'
+
+      Time.strptime(timestamp_param, date_fmt)
```

Comment:
Hokay, so. I wanted to accommodate both timestamp formats (what everything we built is using, and what the IRS is sending). This felt like the easiest way to accommodate that -- just check for a literal dot.

We could do a more elegant regexp, or potentially nested exception handling -- catch ArgumentError and try again with the other format, potentially catching it again, but that felt gross. This seemed least bad, but I wouldn't put this on my resume as a shining example of the best code I've ever written.